### PR TITLE
FIX Don't strip `<header>` tag from `HTMLValue`

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Description

Stops `HTMLValue` from stripping the `<header>` tag.

## Manual testing steps

See linked issue

## Issues

- https://github.com/silverstripe/silverstripe-framework/issues/11298

### CODE REVIEW

File: tests/php/View/Parsers/HTMLValueTest.php [hunk 1]
```diff
@@ -160,4 +160,32 @@ public function testValidHTMLInNoscriptTags()
...
+    {
+        return [
+            [
+                'input' => '<html><head></head><body><div><p>blahblah</p></div></body></html>',
+                'expected' => '<div><p>blahblah</p></div>',
+            ],
+            [
+                'input' => '<html><head></head><body><header></header><div><p>blahblah</p></div></body></html>',
+                'expected' => '<header></header><div><p>blahblah</p></div>',
+            ],
```
Comment (GuySartorelli) [inline tests/php/View/Parsers/HTMLValueTest.php:174]: Confirmed this test failed before this PR.

File: src/View/Parsers/HTMLValue.php [hunk 2]
```diff
@@ -32,7 +32,7 @@ public function __construct($fragment = null)
      */
     public function setContent($content)
     {
-        $content = preg_replace('#</?(html|head|body)[^>]*>#si', '', $content);
+        $content = preg_replace('#</?(html|head|body)(\s[^>]*)?>#si', '', $content);
```
Comment (GuySartorelli) [inline src/View/Parsers/HTMLValue.php]: Went for the more broad approach here rather than the suggestion in the issue - HTML6 could introduce new tags like `<heading>`, or someone could be using a front-end JS library that consumes custom tags that this would otherwise strip.

**Comment by GuySartorelli** [comment]
Prefer-lowest failures are expected - that's happening all over the place. I suspect that's what https://github.com/silverstripe/gha-ci/issues/132 is intended to fix.

**Review by emteknetnz — APPROVED** [review]

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Solution looks fine, though there's failing unit tests

File: src/View/Parsers/HTMLValue.php (hunk 2 above)
Comment (GuySartorelli) [reply src/View/Parsers/HTMLValue.php]: UPDATE: I've reverted back to using the suggestion from the issue because it doesn't have the same CI failure my attempt did. Specifically there was an edge case (which we already have a unit test for, which is what failed) where some really weird invalid HTML could result in multiple `<body>` tags existing in the HTML.

The `HTMLValue` implementation relies on exactly one `<body>` tag being present, so that's a non-starter. People who want custom weird tags will have to name them something else, and we'll just have to be aware of changes when future versions of HTML roll around.

**Comment by GuySartorelli** [comment]
See https://github.com/silverstripe/silverstripe-framework/pull/11302#discussion_r1669476886

**Review by emteknetnz — APPROVED** [review]

File: src/View/Parsers/HTMLValue.php [hunk 3]
```diff
@@ -32,7 +32,7 @@ public function __construct($fragment = null)
      */
     public function setContent($content)
     {
-        $content = preg_replace('#</?(html|head|body)[^>]*>#si', '', $content);
+        $content = preg_replace('#</?(html|head(?!er)|body)[^>]*>#si', '', $content);
```
Comment (michalkleiner) [inline src/View/Parsers/HTMLValue.php:35]: Just a comment here — I know it's cooler to use more regex foo, but imho here it would be much easier for anyone reading the list of tags if it was explicitly listed as `html|head|body|header`.
